WireUI/Sources/WireUIFoundation/Main/MainCoordinator.swift

```diff
@@ -0,0 +1,364 @@
+            splitViewController.newConversation = nil
+            let navigationController = UINavigationController(rootViewController: newConversation)
+            splitViewController.present(navigationController, animated: false)
+        }
+
+        mainSplitViewState = .collapsed
+        conversationList.splitViewInterface = .collapsed
+    }
+
+    public func splitViewControllerDidExpand(_ splitViewController: UISplitViewController) {
+        guard let splitViewController = splitViewController as? SplitViewController,
+              splitViewController === self.splitViewController
+        else { return }
+
+        // move view controllers from the tab bar controller to the supplementary column
+        if tabBarController.selectedContent == .conversations {
+            let conversationViewController = tabBarController.conversations!.conversationList
+            tabBarController.conversations = nil
+            splitViewController.conversationList = conversationViewController
+        }
+
+        // if the archived conversations view controller was visible, present it
+        if tabBarController.selectedContent == .archive {
+            let archive = tabBarController.archive
+            tabBarController.archive = nil
+            splitViewController.archive = archive
+        }
+
+        // if the settings were visible, present it
+        if tabBarController.selectedContent == .settings {
+            let settings = tabBarController.settings
+            tabBarController.settings = nil
+            splitViewController.settings = settings
+        }
+
+        // the new conversation view controller in collapsed mode is
+        // presented in a separate navigation controller
+        if let newConversation {
+            let navigationController = newConversation.navigationController!
+            navigationController.presentingViewController!.dismiss(animated: false)
+            navigationController.viewControllers = []
+            navigationController.view.layoutIfNeeded()
+            splitViewController.newConversation = newConversation
+        }
+
+        mainSplitViewState = .expanded
+        let conversationList = (tabBarController.conversations?.conversationList ?? splitViewController.conversationList)
+        conversationList!.splitViewInterface = .expanded
+    }
+
+    // MARK: - UITabBarControllerDelegate
+
+    public func tabBarController(_ tabBarController: UITabBarController, didSelect viewController: UIViewController) {
+        guard let tabBarController = tabBarController as? TabBarController,
+              tabBarController === self.tabBarController
+        else { return }
+
+        switch tabBarController.selectedContent {
+        case .contacts, .folders:
+            break // `.contacts` and `.folders` are removed for navigation overhaul
+
+        case .conversations:
+            switch conversationList.conversationFilter?.map() {
+            case .none:
```

Comment:
Optional: I think there is duplication with earlier in the file. Consider extracting this logic to a private helper.

Reply:
Fixed, thanks!